server.js

```diff
@@ -14,10 +21,68 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const WMessage = [welcomeMessage];
 
+// create a message and post it to database
+app.post("/messages", (req, res) => {
+  let userName = req.body.userName;
+  let userMessage = req.body.userMessage;
+  if (userName === "" || userMessage === "") {
+    res.status(400).send(`Invalid:No userName & userMessage`);
+  } else {
+    const newChat = {
+      id: ++maxId,
+      from: userName,
+      text: userMessage,
+    };
+    data.push(newChat);
+    save();
+    res.send(data);
+  }
+});
+
+// serves the main HTML file for the home page
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// give all the messages in the dataBase
+app.get("/messages/search", (req, res) => {
+  let userSearchId = req.query.randomNumber;
+  if (!Number(userSearchId)) {
```

Comment:
What does `Number` return if the input is not a number? How should we check for that return value?